### DISCUSSION TITLE
How to add distributed tracing support (OTEL) to DynamoDBTemplate?

### DISCUSSION THREAD

**Discussion by jpOlivo-uala** [awspring/spring-cloud-aws#1244] [Q&A] [answered]
I saw some discussions about add distributed tracing support to SqsTemplate but I do not found some related with DynamoDBTemplate

https://github.com/awspring/spring-cloud-aws/discussions/902
https://github.com/awspring/spring-cloud-aws/issues/646
https://github.com/aws/aws-sdk-java-v2/issues/4611

I want know how if is possible add tracing support (OTEL) to DynamoDBTemplate, also if exists some approach that i can follow can implement this in the meantime

Thanks and Regards,

**Comment by maciejwalkowiak** [comment, marked as answer]
Until support for dynamodb template is in place, perhaps instrumenting AWS SDK is good enough for you? Here are the docs I found for OTEL https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/aws-sdk/aws-sdk-2.2/library/README.md
